scripts/Inelastic/CrystalField/function.py

```diff
@@ -489,3 +489,105 @@ def update(self, func1, func2=None):
             self.background.update(func1)
         else:
             self.peak.update(func1)
+
+
+class ResolutionModel:
+    """
+    Encapsulates a resolution model.
+    """
+    default_accuracy = 1e-4
+    max_model_size = 100
+
+    def __init__(self, model, xstart=None, xend=None, accuracy=None):
+        """
+        Initialize the model.
+
+        :param model: Either a prepared model or a single python function or a list
+                    of functions. If it's functions they must have signatures:
+                        func(x: ndarray) -> ndarray
+                    A prepared model is a tuple of exactly two arrays of floats of
+                    equal sizes or a list of such tuples. The first array in the tuple
+                    is the x-values and the second array is the y-values of the resolution
+                    model.
+        :param xstart:
+        :param xend:
+        :param accuracy: (Optional) If given and model argument contains functions it's used
+                    to tabulate the functions such that linear interpolation between the
+                    tabulated points has this accuracy. If not given a default value is used.
+        """
+        from inspect import isfunction
+        self.multi = False
+        if isfunction(model):
```

Comment:
Could we change this check (and also in line 527) to `isfunction(model) or ismethod(model)`? Because, I would also like to pass the bound method `getResolution` of the `PyChop2` object to `ResolutionModel`, without using a `lambda` - at the moment you have to do:

```
from PyChop import PyChop2
from CrystalField import CrystalField, ResolutionModel
chop_res = PyChop2('MARI', 'S', 250)
chop_res.setEi(40)
res_mod = ResolutionModel(lambda x: chop_res.getResolution(x), xstart=-10, xend=35, accuracy=0.1)
cf = CrystalField('Ce', 'C4v', B20=0.451, B40=0.025, B44=0.317, Temperature=4, 
                  ResolutionModel=res_mod, WidthVariation=0.1)
```

whereas it would be nicer for the user if:

```
res_mod = ResolutionModel(chop_res.getResolution, xstart=-10, xend=35, accuracy=0.1)
```

worked.